patches/net/minecraft/data/recipes/ShapedRecipeBuilder.java.patch

```diff
@@ -0,0 +1,47 @@
+--- a/net/minecraft/data/recipes/ShapedRecipeBuilder.java
++++ b/net/minecraft/data/recipes/ShapedRecipeBuilder.java
+@@ -25,6 +_,7 @@
+     private final RecipeCategory category;
+     private final Item result;
+     private final int count;
++    private final ItemStack resultStack; // Neo: add stack result support
+     private final List<String> rows = Lists.newArrayList();
+     private final Map<Character, Ingredient> key = Maps.newLinkedHashMap();
+     private final Map<String, Criterion<?>> criteria = new LinkedHashMap<>();
+@@ -33,9 +_,14 @@
+     private boolean showNotification = true;
+ 
+     public ShapedRecipeBuilder(RecipeCategory p_249996_, ItemLike p_251475_, int p_248948_) {
++        this(p_249996_, new ItemStack(p_251475_, p_248948_));
++    }
++
++    public ShapedRecipeBuilder(RecipeCategory p_249996_, ItemStack result) {
+         this.category = p_249996_;
+-        this.result = p_251475_.asItem();
+-        this.count = p_248948_;
++        this.result = result.getItem();
++        this.count = result.getCount();
++        this.resultStack = result;
+     }
+ 
+     public static ShapedRecipeBuilder shaped(RecipeCategory p_250853_, ItemLike p_249747_) {
+@@ -46,6 +_,10 @@
+         return new ShapedRecipeBuilder(p_251325_, p_250636_, p_249081_);
+     }
+ 
++    public static ShapedRecipeBuilder shaped(RecipeCategory p_251325_, ItemStack result) {
```

Comment:
Given the fact this is a totally new method and isn't overlapping the existing one for the body we should probably give the category parameter a proper name here rather than a srg name.

Reply:
My problem is then how we will update these names in the future. As long as the params remain unnamed, it is very easy to update parchment to use them. (Just string search for the unmapped names)